delocate/libsana.py

```diff
@@ -48,16 +49,24 @@ def tree_libs(start_path, filt_func=None):
     * http://matthew-brett.github.io/pydagogue/mac_runtime_link.html
     """
     lib_dict = {}
+    env_var_paths = get_environment_variable_paths()
     for dirpath, dirnames, basenames in os.walk(start_path):
         for base in basenames:
             depending_libpath = realpath(pjoin(dirpath, base))
             if not filt_func is None and not filt_func(depending_libpath):
                 continue
             rpaths = get_rpaths(depending_libpath)
+            search_paths = rpaths + env_var_paths
             for install_name in get_install_names(depending_libpath):
-                lib_path = (install_name if install_name.startswith('@')
-                            else realpath(install_name))
-                lib_path = resolve_rpath(lib_path, rpaths)
+                # If the library starts with '@rpath' we'll try and resolve it
+                # We'll do nothing to other '@'-paths
+                # Otherwise we'll search for the library using env variables
+                if install_name.startswith('@rpath'):
+                    lib_path = resolve_rpath(install_name, search_paths)
```

Comment:
Sorry - just a final check - it's true, then that if the name is `@rpath/foo.dylib`, macOS will search not only the rpaths specified in the list of rpaths from the binary, but also paths on `DYLD_LIBRARY_PATH` etc?